### PR TITLE
CDAP-1210 Client and CLI for adapters

### PR DESCRIPTION
**Pull request by jwang47**

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java [hunk 1]
```diff
@@ -101,7 +101,7 @@
 
   @Inject
   public DefaultStore(CConfiguration conf,
-                      LocationFactory locationFactory,
+                     LocationFactory locationFactory,
```
Comment (gokulavasan) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: alignment

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateAdapterCommand.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/CreateAdapterCommand.java]: 2015

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/DeleteAdapterCommand.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/DeleteAdapterCommand.java]: 2015

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/ListAdaptersCommand.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/ListAdaptersCommand.java]: 2015

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/UseNamespaceCommand.java]: 2015

File: cdap-client-tests/src/test/java/co/cask/cdap/client/AdapterClientTestRun.java [hunk 6]
```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright © 2014 Cask Data, Inc.
```
Comment (sreevatsanraman) [inline cdap-client-tests/src/test/java/co/cask/cdap/client/AdapterClientTestRun.java]: check all files, yeah in copyright should be 2015

File: cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    public Source(String name, Map<String, String> properties) {
+      this.name = name;
+      this.properties = properties;
+    }
+  }
+
+  /**
+   *
```
Comment (sreevatsanraman) [inline cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java]: Doc

File: cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+  public Source source;
+  public Sink sink;
+
+  public String getType() {
+    return type;
+  }
+
+  /**
+   *
```
Comment (sreevatsanraman) [inline cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java]: Doc

File: cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+    public Sink() {
+    }
+
+    public Sink(String name, Map<String, String> properties) {
+      this.name = name;
+      this.properties = properties;
+    }
+  }
+
+  public AdapterSpecification toAdapterSpec(String name, co.cask.cdap.proto.Source.Type sourceType,
```
Comment (sreevatsanraman) [inline cdap-proto/src/main/java/co/cask/cdap/proto/AdapterConfig.java]: Doc for public method.

File: cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java [hunk 10]
```diff
@@ -88,6 +88,31 @@ public URL resolveURL(String path) throws MalformedURLException {
...
+   * @return URL of the resolved path
+   * @throws MalformedURLException
+   */
+  public URL resolveURL(String apiVersion, String namespace, String path) throws MalformedURLException {
+    return getBaseURI().resolve("/" + apiVersion + "/namespaces/" + namespace + "/" + path).toURL();
+  }
+
+  /**
+   * Resolves a path against the target CDAP server with the provided apiVersion
+   * @param apiVersion the api version to use
```
Comment (sreevatsanraman) [inline cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java]: New line before param to keep the style consistent

**Comment by sreevatsanraman** [comment]
Fix the style issues and copyright, then merge. LGTM

File: cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java (hunk 10 above)
Comment (jwang47) [inline cdap-client/src/main/java/co/cask/cdap/client/config/ClientConfig.java]: Oh yeah, actually newline before param is required to make it valid javadoc.

**Pull request merged**